projects/plugins/jetpack/json-endpoints/class.wpcom-json-api-list-users-endpoint.php

```diff
@@ -229,12 +229,23 @@ function ( $viewer ) use ( $args ) {
 
 					$combined_users = array_merge( $users, $viewers );
 
-					// When viewers are included, we ignore the order & orderby parameters.
+					// When viewers are included, we try to respect orderby if it's login or email.
 					if ( $include_viewers ) {
+						// First we need to be sure to obtain the viewer emails and logins.
+						foreach ( $combined_users as &$user ) {
+							$wp_user     = new WP_User( $user->ID );
+							$user->email = $wp_user->user_email;
+							$user->login = $wp_user->user_login;
+						}
+
+						$order   = strtolower( $args['order'] ) ?? 'asc';
+						$orderby = strtolower( $args['order_by'] ) ?? 'name';
```

Comment:
I don't get this error? Why would the right hand side be unnecessary? `$args['order']` could well be an empty string.

Reply:
Apologies if this comment isn't helpful (it's my first review comment in 3 months so my confidence is quite low 😄). As I understand it, the `strtolower()` function will always return a string (even an empty string), and the null coalescing operator will only return the right operand if the left is `null` (an empty string will result in the left operand winning).

In this case, would a ternary work? Something like:

```php
$orderby = ! empty( $args['order_by'] ) ? strtolower( $args['order_by'] ) : 'name';
```

Reply:
Good catch. I'm pretty sure, for example, 'asc' will never actually be assigned to `$order`, because the null coalescing operator will use the empty string from `strtolower` (assuming `$args['order']` exists)

Reply:
Oh I see that makes sense! Yes I think in this case we'll have to use a ternary.